core/lib/zksync_core/src/state_keeper/mempool_actor.rs

```diff
@@ -77,7 +80,18 @@ impl<G: L1GasPriceProvider> MempoolFetcher<G> {
             let latency = KEEPER_METRICS.mempool_sync.start();
             let mut storage = pool.access_storage_tagged("state_keeper").await.unwrap();
             let mempool_info = self.mempool.get_mempool_info();
-            let l2_tx_filter = l2_tx_filter(self.l1_gas_price_provider.as_ref(), fair_l2_gas_price);
+
+            let protocol_version = storage
+                .protocol_versions_dal()
+                .base_system_contracts_by_timestamp(seconds_since_epoch())
+                .await
+                .1;
+
+            let l2_tx_filter = l2_tx_filter(
+                self.l1_gas_price_provider.as_ref(),
+                fair_l2_gas_price,
+                protocol_version.into(),
+            );
```

Comment:
The more I read it, seems we need a kinda Oracle with the current protocol version of the system. 
@popzxc wdyt?

Reply:
What do you mean? Some kind of in-memory store to avoid accessing DAL?
Probably it's a nice-to-have, but generally, it seems to be a premature optimization.

Reply:
Do we have a struct that represents every option/config parameter that can differ between VM implementations - obviously, apart from the VM itself which lives in zk_evm repo? Do I understand it right that currently to update VM version we need to do:
* everything otherwise needed to update protocol version
* reference corresponding tags from crypto deps in our cargo.toml
* implement multivm-related methods (or copy/use the implementaiton of the previous version) - such as the ones that this PR introduces

Is there a way to get an overview of all the particularities related to a particular VM version?

Reply:
in the database we have this struct 
<img width="508" alt="image" src="https://github.com/matter-labs/zksync-era/assets/8031374/87476a02-25fd-4344-b404-4ed41f2fb8f3">

Normally it represents everything.

Reply:
@popzxc It's a bit scary part right here. 
because we get the protocol version, not from the latest miniblock, but from the timestamp. 

The timestamp is not when we are doing an upgrade, but when we can start an upgrade and the upgrade will start with the next batch. 

So, it have to track somehow for sure what is the current protocol version of the system

Reply:
Hm, agree. It does sound kinda dangerous.

Reply:
@Deniallugo this is mempool fetcher that usually retrieves transactions that must not need to comply with the parameters of the current batch (for instance, it re-queries L1 gas price all the time). I can change the logic to query the protocol version from the latest miniblock. The worst thing that can happen is that the transactions won't be picked up for some very short period of time after which the server will be updated and everything will work as needed

Reply:
@StanislavBreadless I think it's better to query protocol version of the latest miniblock, unless fee formulas will be quite different and there is a risk of getting stuck

Reply:
@perekopskiy they will be different, but we'll try to make the transition as smooth as possible